### ISSUE TITLE
Is there a Discussion Board? 

### ISSUE THREAD

**Issue by aoirthoir** [joelesko/tht#16]
I just stumbled on THT and love it. Is there a discussion board someplace to keep people like me with discussion ideas from clogging the issues with questions? (Note added: I see there is a discord server and twitter. But could not find discussions on github)

Is there a single download PDF of all instructions, and if not, would you mind if I converted the entire website to such a PDF?

So far everything I see I like. 

I am curious about the SQL injection if it does something similar to these:

[MeekroDB](https://meekro.com/docs/retrieving-data.html)

[PHPDelusions SQL Injkection](https://phpdelusions.net/sql_injection)

[SafeMySQL](https://phpdelusions.net/safemysql)

As to structures, with the curly braces, is it possible to use END instead of curly braces?

```
if something
  do this
  if something else
    Do more this
  end
end
```

rather than
```

if something {
  do this
  if somethign else {
    do more this
  }
}
```
And the same for fn, tm, loop, foreach etc.

If it is not possible, would you be willing to add the capability?

I do realize a lot of people love the curly braces, whereas I like specific ends...

I DO LOVE no semicolons.. that is awesome... I also LOVE lists starting at 1 rather than zero...

I also LOVE that IF does not have to be enclosed in parens if (something) {}. 

Basically what you have here is just what I have been looking for in my complaints about php...i have to study your manual more. 

Thanks for this great work. If there is a message or discussion area or if you would be willing to turn it on here on github, I will go there from now on.

Also the jcon website jconformat.org is not showing up, I presume it is decrepcated? the website not the software. And JCON is awesome too! I like how cleanly you think about code.

**Comment by joelesko** [comment]
Hi [aoirthoir](https://github.com/aoirthoir),

I just opened up the Discussions tab.  Thanks for suggesting this!

To answer your other questions:

1) Sure, you can convert it to PDF for your own use.  

2) SQL.  yes and no. THT is more secure because it requires the query template string to be a hard-coded/compile-time literal.   At run-time it uses placeholders just like most SQL libraries. 

3) End.  Possible, yes, but not something I would do.  A key part of the design is to be consistency.  I have considered an "outline" mode that lets you use indentation like Python, but right now consistency is more important. 

4) Thanks for the heads up on the jcon site.  I took another look at HJSON recently, and realized that it might better to use that instead, given it has more support.  Open to thoughts on that. 

Thanks!

**Comment by aoirthoir** [comment]
1. Thank you. I am in the process of combining your entire website to one large file so I can print to PDF. Then I will finagle format a bit more. It wont look super fance but it will be usable. Later I can work on a more official PDF if you want.

2. Regarding SQL, yes that is how those libraries work also. Except THT, doesnt just work that way with SQL. You manage all injection which is awesome. I also like your db module which lets me use THT commands that work in all the Databases (I presume it works in them all).

3. I can has sads? Lol. I do prefer end statements to the {} brackets as they always seemed like noise to me. But then I come from a system (COBOL) that had almost no noise, unlike C and PHP. But I will survive. Just because I prefer something does not mean it has to be. Considering all the other enhancements THT gives me, I can live with squiggly brackets.

As to Python style indentation, it is not the worse thing by far, I just prefer something more explicit. Whether it is END or {}. 

4. I took a look at HJSON today and I do like it. I will give more feedback. One thing I liked about it was the ability to just use strings without quotes. At least in certain cases.

I'll reserve the rest of my comments to the discussion area, unless its an actual error that I notice. I tried to install on Dreamhost so I will create an issue for that.

Thanks for your great work. THT is exactly the kind of thing I was looking for and frankly the kind of thing we need in web development.
